pkgs/build-support/docker/default.nix

```diff
@@ -259,13 +259,14 @@ rec {
 
             mkdir -p image/$extractionID/layer
             tar -C image/$extractionID/layer -xpf image/$layerTar
-            rm image/$layerTar
```

Comment:
We could keep the max disk space low by keeping this `rm`.

What if we preprocess the list instead?

Here's a filter we could apply to the `layer-list` file

```shell
function dedup() {
  declare -A seen
  while read ln; do
    if [[ -z "${seen["$ln"]:-}" ]]; then
      echo "$ln"; seen["$ln"]=1
    fi
  done
}
```

Use tac to keep the last instead of the first.

```console
$ (echo 1; echo a; echo 2; echo a; echo 3;) | tac | dedup | tac
1
2
a
3
```

This might even be more consistent with docker overlayfs behavior if `tar` were to not entirely replace some fs nodes. So not sure if that applies, but if it does, it's another benefit.